dask/blockwise.py

```diff
@@ -517,6 +455,195 @@ def cull(self, keys, all_hlg_keys):
+    output_indices: tuple
+        The output indices, like ``('i', 'j', 'k')`` used to determine the
+        structure of the block computations
+    dsk: dict
+        A small graph to apply per-output-block.  May include keys from the
+        input indices.
+    indices: Tuple[str, Tuple[str, str]]
+        An ordered mapping from input key name, like ``'x'``
+        to input indices, like ``('i', 'j')``
+        Or includes literals, which have ``None`` for an index value
+    numblocks: Dict[key, Sequence[int]]
+        Number of blocks along each dimension for each input
+    concatenate: boolean
+        Whether or not to pass contracted dimensions as a list of inputs or a
+        single input to the block function
+    new_axes: Dict
+        New index dimensions that may have been created, and their extent
+    output_blocks: Set[Tuple]
+        Specify a specific set of required output blocks. Since the graph
+        will only contain the necessary tasks to generate these outputs,
+        this kwarg can be used to "cull" the abstract layer (without needing
+        to materialize the low-level graph).
+
+    See Also
+    --------
+    dask.blockwise.BlockwiseIO
+    """
+
+    def __init__(
+        self,
+        io_subgraph,
+        output,
+        output_indices,
+        dsk,
+        indices,
+        numblocks,
+        concatenate=None,
+        new_axes=None,
+        output_blocks=None,
+    ):
+        # Handle the case that this is a "pure" IO layer (dsk is None).
+        # Note that `dsk` should only be defined after fusion.
+        if not dsk:
+            # Extract actual IO function for SubgraphCallable construction.
+            # Wrap func in `PackedFunctionCall`, since it will receive
+            # all arguments as a sigle (packed) tuple at run time.
+            subgraph = io_subgraph[1]
+            if subgraph:
+                # We assume a 1-to-1 mapping between keys (i.e. tasks) and
+                # chunks/partitions in `io_subgraph`, and assume the first
+                # (callable) element is the same for all tasks.
+                any_key = next(iter(subgraph))
+                io_func = subgraph.get(any_key)[0]
+            else:
+                io_func = None
+            ninds = 1 if isinstance(output_indices, str) else len(output_indices)
+            dsk = {
+                output: (
+                    PackedFunctionCall(io_func),
+                    *[blockwise_token(i) for i in range(ninds)],
+                )
+            }
+
+        # Super-class initializer
+        super().__init__(
+            output,
+            output_indices,
+            dsk,
+            indices,
+            numblocks,
+            concatenate=None,
+            new_axes=None,
+            output_blocks=None,
+        )
+
+        # BlockwiseIO requires an `io_subgraph` input.
+        # This is a tuple: (key-name, subgraph)
+        self.io_name, self.io_subgraph = io_subgraph
```

Comment:
Not sure if it makes sense to continue accepting a `(name, io_subgraph)` tuple in the `BlockwiseIO` initializer.  I personally like this, but also recognize that two separate args may be more intuitive.

Reply:
I think two separate args will be more intuitive

Reply:
yeah - I'm starting to come to the same conclusion. I'll see what I can do.

Reply:
Okay - I modified the code to use two separate args.